ogre2/src/Ogre2GpuRays.cc

```diff
@@ -781,6 +781,13 @@ void Ogre2GpuRays::UpdateSampleTexture()
       auto [x, y] = points[index];
       x = hmin + (x + 1) / 2 * (hmax - hmin);
       y = vmin + (y + 1) / 2 * (vmax - vmin);
+
```

Comment:
We only need to compute the points above if the scanning pattern is avia. Let's skip over that if we are in rasterization mode.

Reply:
I don't think we can do that since direction setting was taken out from `CreateSampleTexture`, if we don't run `UpdateSampleTexture` there is no point cloud in rviz. The only way I can come up now is to take the old code back.

Reply:
Yeah, we can keep the `UpdateSampleTexture`. We can simply skip over line 767 and not compute the points, since we aren't using them. Same with lines 781-783. `x` and `y` will just be `float` in that case.

Reply:
What does it mean by ``` x and y will just be float in that case.```

Reply:
Currently we are defining x and y automatically by getting a point from the point list. Instead we can just do `float x` and `float y`.